Improve README.md and related files (#484)
HumanCompatibleAI/imitation: docs/development/contributing/ways-to-contribute.rst

Ways to contribute
==================

There are four main ways you can contribute to imitation:

> * `Reporting bugs <#reporting-bugs>`_

Reviewer: These hyperlinks seem broken (anchors are #id1, #id2, etc not #reporting-bugs, #suggesting-new-features in https://imitation--484.org.readthedocs.build/en/484/development/contributing/ways-to-contribute.html)

* `Suggesting new features <#suggesting-new-features>`_
* `Contributing to the documentation <#contributing-to-the-documentation>`_
* `Contributing to the codebase <#contributing-to-the-codebase>`_

> Please, note that by contributing to the project, you are agreeing to license your work under :ref:`imitation's license <License>`, as per

Reviewer: ```suggestion
Please note that by contributing to the project, you are agreeing to license your work under :ref:`imitation's MIT license <License>`, as per
```

You can submit a new bug report by creating an issue on `GitHub <https://github.com/HumanCompatibleAI/imitation/issues/new>`_ and labeling it as an *enhancement*. **Before you do so, please make sure that**\ :

* You have checked the `documentation <https://imitation.readthedocs.io/en/latest/>`_ that relates to your request, as it may be that such feature is already available,
* You have checked `existing feature requests <https://github.com/HumanCompatibleAI/imitation/issues?q=is%3Aissue+label%3Aenhancement+is%3Aopen+>`_ to make sure that there is no similar request already under discussion, and
* You have a minimal use case that describes the relevance of the feature.

When you **submit the feature request**:

* Use a clear and descriptive title for the GitHub issue to easily identify the suggestion.
* Describe the current behavior, and explain what behavior you expected to see instead and why.
* If you want to request an API change, provide examples of how the feature would be used.
* If you want to request a new algorithm implementation, please provide a link to the relevant paper or publication.

Contributing to the documentation
---------------------------------

> One of the simplest ways to get started contributing imitation is through improving the documentation. Currently, our documentation some gaps, and we would love to have you help us fill them. You can help by adding missing sections of the API docs, editing existing content to make it more readable, clear and accessible, or contribute with new content, such as tutorials and FAQs.

Reviewer: ```suggestion
One of the simplest ways to start contributing to imitation is through improving the documentation. Currently, our documentation has some gaps, and we would love to have you help us fill them. You can help by adding missing sections of the API docs, editing existing content to make it more readable, clear and accessible, or contributing new content, such as tutorials and FAQs.
```

> If you have struggled to understand something about our codebase and managed to figure it out in the end, please consider improving the relevant documentation section, adding a tutorial or a FAQ entry, so that other users can learn from your experience.

Reviewer: ```suggestion
If you have struggled to understand something about our codebase and managed to figure it out in the end, please consider improving the relevant documentation section, or adding a tutorial or a FAQ entry, so that other users can learn from your experience.
```

Before submitting a pull request, please create an issue with the *documentation* label so that we can track the gap. You can then reference the issue in your pull request by including the issue number.

Contributing to the codebase
----------------------------

You can contribute to the codebase by proposing solutions to issues or feature suggestions you've raised yourself, or selecting an existing issue to work on. Please, make sure to create an issue on GitHub before you start working on a pull request, as explained in `Reporting bugs <#reporting-bugs>`_ and `Suggesting new features <#suggesting-new-features>`_.

Once you're ready to start working on your pull request, please make sure to follow our **coding style guidelines**\ :

* PEP8, with line width 88.
* Use the ``black`` autoformatter.
* Follow the `Google Python Style Guide <http://google.github.io/styleguide/pyguide.html>`_ unless
  it conflicts with the above. Examples of Google-style docstrings can be found
  `here <https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html>`_.

**Before you submit**\ , please make sure that:

> * PRs should include unit tests for any new features, and add type annotations where possible. It is OK to omit annotations when it would make the code significantly more complex.

Reviewer: ```suggestion
* Your PR includes unit tests for any new features.
* Your PR includes type annotations, except when it would make the code significantly more complex.
```
